traitsui/tests/editors/test_instance_editor.py

```diff
@@ -29,50 +30,54 @@ def get_view(style):
 
-    def check_editor_update_value(self, style):
+    def test_simple_editor(self):
         obj = NonmodalInstanceEditor()
         tester = UITester()
-        with tester.create_ui(obj, dict(view=get_view(style))) as ui:
+        with tester.create_ui(obj, dict(view=get_view("simple"))) as ui:
             instance = tester.find_by_name(ui, "inst")
-            editor = instance.locate(locator.NestedUI())
-            value_txt = editor.find_by_name("value")
+            instance.perform(command.MouseClick())
+            value_txt = instance.find_by_name("value")
             value_txt.perform(command.KeySequence("abc"))
-            if style == 'simple':
-                # the below code requires ui_base implementation to work
-                # ok_button = editor.find_by_id("OK")
-                # ok_button.perform(command.MouseClick())
-                press_ok_button(editor.target)
             self.assertEqual(obj.inst.value, "abc")
 
-    def test_simple_editor(self):
-        return self.check_editor_update_value('simple')
-
     def test_custom_editor(self):
-        return self.check_editor_update_value('custom')
+        obj = NonmodalInstanceEditor()
+        tester = UITester()
+        with tester.create_ui(obj, dict(view=get_view("custom"))) as ui:
+            value_txt = tester.find_by_name(ui, "inst").find_by_name("value")
+            value_txt.perform(command.KeySequence("abc"))
+            self.assertEqual(obj.inst.value, "abc")
 
-    def check_resynch_editor(self, style):
+    def test_custom_editor_resynch_editor(self):
         edited_inst = EditedInstance(value='hello')
         obj = NonmodalInstanceEditor(inst=edited_inst)
         tester = UITester()
-        with tester.create_ui(obj, dict(view=get_view(style))) as ui:
-            instance = tester.find_by_name(ui, "inst")
-            editor = instance.locate(locator.NestedUI())
-            value_txt = editor.find_by_name("value")
+        with tester.create_ui(obj, dict(view=get_view("custom"))) as ui:
+            value_txt = tester.find_by_name(ui, "inst").find_by_name("value")
             displayed = value_txt.inspect(query.DisplayedText())
             self.assertEqual(displayed, "hello")
             edited_inst.value = "bye"
             displayed = value_txt.inspect(query.DisplayedText())
             self.assertEqual(displayed, "bye")
 
-    def test_custom_editor_resynch_editor(self):
-        return self.check_resynch_editor('custom')
-
     def test_simple_editor_resynch_editor(self):
-        return self.check_resynch_editor('simple')
+        edited_inst = EditedInstance(value='hello')
+        obj = NonmodalInstanceEditor(inst=edited_inst)
+        tester = UITester()
+        with tester.create_ui(obj, dict(view=get_view("simple"))) as ui:
+            instance = tester.find_by_name(ui, "inst")
+            instance.perform(command.MouseClick())
+
+            value_txt = tester.find_by_name(ui, "inst").find_by_name("value")
```

Comment:
could we do `value_txt =instance.find_by_name("value")` here?

Reply:
Yes we can!